src/main/perf/facets/IndexFacets.java

```diff
@@ -124,4 +131,20 @@ public static void main(String args[]) throws IOException {
         AtomicInteger counter = new AtomicInteger(0);
         return item -> consumer.accept(counter.getAndIncrement(), item);
     }
+
+    private static void deleteDirectoryIfExists(Path path) throws IOException {
+        if (Files.notExists(path) == false) {
+            try(Stream<Path> walk = Files.walk(path)) {
+                walk.sorted(Comparator.reverseOrder()).forEach(IndexFacets::deleteFile);
+            }
+        }
+    }
+
+    private static void deleteFile(Path path) {
+        try {
+            Files.delete(path);
+        } catch (IOException e) {
+            System.err.printf("Unable to delete this path : %s%n%s", path, e);
```

Comment:
How about re-throwing the `IOException` so the whole process fails/exits?

Reply:
Changed in next revision.